Update README with ssh tunnel instructions

ijdickinson opened the pull request:
Access to the data API on the new infrastructure is different than it
was on the old system. After some help from @mika018 (thanks Mihajlo!) I
was able to get an ssh tunnel working correctly.

These instructions work on Linux (Ubuntu 21.04). They may need amending
and tweaking to cover MacOS and Windows.

ijdickinson committed:
Update README with ssh tunnel instructions

Access to the data API on the new infrastructure is different than it
was on the old system. After some help from @mika018 (thanks Mihajlo!) I
was able to get an ssh tunnel working correctly.

These instructions work on Linux (Ubuntu 21.04). They may need amending
and tweaking to cover MacOS and Windows.

joescottdave commented on README.md:
````diff
@@ -28,13 +28,96 @@ To install on dev machine:
+$
+```
+
+Assuming this succeeds, you can set up an ssh tunnel to map the port where the data
````
For the completely uninitiated who are following this might be worth throwing in a line about what "this works" would look like and then exiting the shell?

ijdickinson replied:
Thanks. I've added the sample output, and a note about shutting down the ssh process at the end as well.

andrew-pickin-epi replied:
NO.

Please stop baking ssh tunnels to the way we work.
If their is a need to access a host then add a requirement and let work the problem properly.
Either use the VPN or let's fix the SGs

joescottdave reviewed (commented):
I've followed along with this, I've made one suggestion where I can see an earlier version of me following along just fine and then proceeding to enter the remaining commands into the shell of the ssh'd machine.

mika018 commented on README.md:
````diff
@@ -28,13 +28,96 @@ To install on dev machine:
+COMMAND   PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
+ssh     71358  ian    3u  IPv6 766085      0t0  TCP ip6-localhost:http-alt (LISTEN)
+ssh     71358  ian    6u  IPv4 766086      0t0  TCP localhost:http-alt (LISTEN)
+```
````
Maybe add a line to remind users to close the connection after they are done using it by killing the background process using it's PID.
`kill 71358` in this example.

ijdickinson replied:
Good suggestion. I've added some content to that effect

mika018 commented on README.md:
````diff
@@ -28,13 +28,96 @@ To install on dev machine:
+Check that you can access a server by directly connecting via ssh:
+
+```sh
+$ ssh hmlr_dev_pres_071
````
The server name (or id - 071 in this case) is variable and will not always be 071. I think it updates every time we upgrade the server but don't quote me on this, it's a question for @andrew-pickin-epi.
You can use sensu to find out which entities run in which environment, here's a link for dev: https://sensu-hmlr.epimorphics.net/c/~/n/dev/entities. I would suggest adding an extra step prior to this one for determining the right server name using the link above.

ijdickinson replied:
Good catch, thanks @mika018. I've added a section to address this

andrew-pickin-epi replied:
I'd prefer not to do it this way. 
Way not just access the API LB directly? Then you don't need to know the name of the host.

If this is currently blocked but security groups, it can be in dev, and maybe even via the VPN for other environments.

I would request that in future folk don't go off down ssh tunnel route but discuss their requirements so less ad hoc solutions can be implemented.

mika018 reviewed (commented):
Looks good :+1: 
I added a couple of suggestions.

ijdickinson committed:
Add instructions for determining the hostname

This commit addresses a comment by @mika018: the identity of the server
hosting the API may occasionally change, so it will be necessary to
check the correct host identity for these instructions to be sure of
working.

ijdickinson committed:
Add reminder to kill the SSH tunnel process

This commit addresses a comment from @mika018 to kill the SSH tunnel
process when no longer needed.

ijdickinson committed:
Add sample output from ssh command

This commit addresses a suggestion by @joescottdave to show the output
from the ssh tunnel command.

mika018 commented:
All looks good now from my point of view :+1:

andrew-pickin-epi commented on README.md:
```diff
@@ -28,13 +28,96 @@ To install on dev machine:
+to be coming via localhost on the remote server. We can do this by creating
+an ssh tunnel.
+
+#### Setting up an ssh tunnel to access the data service
```
Please don't make this offical.

SSH access to LR should reference not duplicate the wiki

ijdickinson commented:
As has been discussed on Slack, we are planning to have a meeting to discuss better alternatives, and then document the preferred pattern. In the meantime, I am closing this PR.

Captured as epimorphics/hmlr-linked-data#75
